### PR TITLE
Reduced code size be removing the options "grid_type" and "dord4" con…

### PR DESCRIPTION
**Pull request by middlecoff**
## Proposed changes
Reduce code size be removing the options `grid_type` and `dord4` controlled by the hardwired parameters:
```
integer, parameter :: grid_type = 0
logical, parameter :: dord4 =.true.
```
## Types of changes

What types of changes does your code introduce?
_Put an `x` in all the boxes that apply_

- [ ] Bugfix (Change which fixes an issue)
- [ ] New feature (Change which adds functionality)
- [x] Enhancement to existing functionality (non-breaking change which improves existing code or documentation)
- [ ] Breaking change (fix or feature that changes results and requires updates to test baselines)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/CONTRIBUTING.md) document
- [x] I have ensured all my changes contribute to a common purpose
- [x] I have verified my change does not add debug code
- [x] I have verified my change does not add code that is commented out
- [x] I have verified unit tests pass locally with my changes
- [x] I have verified my changes adhere to style guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: ref/src/sw_core.F90 [hunk 1]
```diff
@@ -631,48 +549,45 @@ subroutine d2a2c_vect(sw_corner, se_corner, ne_corner, nw_corner,     &
...
+    endif
 
-      if ( (je + 1) == npy .or. jed >= (npy - npt) ) then
-        do j = npy-npt+1, jed
-          do i = isd, ied
-            utmp(i, j) = 0.5 * (u(i, j) + u(i, j+1))
-            vtmp(i, j) = 0.5 * (v(i, j) + v(i+1,j) )
-          enddo
+    if ( (je + 1) == npy .or. jed >= (npy - 4) ) then
+      do j = npy-3, jed
```
Comment (robgonzalezpita) [inline ref/src/sw_core.F90:562]: @middlecoff I could be mistaken here, but if you're replacing `npt` with 4, shouldn't this line read `do j = npy - 5, jed`

Comment (middlecoff) [reply ref/src/sw_core.F90:562]: Isn't npy-npt+1 = npy-4+1 = npy-3?

Comment (robgonzalezpita) [reply ref/src/sw_core.F90:562]: You're right, it is. I don't know what I was thinking.

**Review by robgpita — APPROVED** [review]

**Review by christopherwharrop-noaa — APPROVED** [review]
The removal of the `grid_type` and `dord4` here is straightforward and a careful examination shows no issues.

**Pull request merged**